torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -1168,7 +1166,33 @@ class ShapePropagator {
 
     // Requirements:
     //   dims           : 0
-    //   scalar type    : dtype if specified, else preserved
+    //   scalar type    : promoted to floating if integral, else preserved
+    //   device         : preserved
+    //   tensor inputs  : 1
+    //   tensor outputs : 1
+    // Additionally:
+    //   - First input should be the only tensor input
+    static const register_formula_for floating_reduce_ops {
+      {
+        "aten::std(Tensor self, bool unbiased) -> Tensor",
+        "aten::var(Tensor self, bool unbiased) -> Tensor",
+      },
+      [](Node* node) -> type_vec_t {
+        if (auto type = node->input(0)->type()->cast<TensorType>()) {
+          auto ret = type->withDim(0);
+          if (isIntegralType(*(type->scalarType()), /*includeBool=*/true)) {
+            const auto default_dtype = at::typeMetaToScalarType(caffe2::get_default_dtype());
```

Comment:
i think you want `c10::get_default_type` or `at::get_default_dtype` but not `caffe2`.

Reply:
Sounds good. Thanks @eellison!